### ISSUE TITLE
[Rule Request] Deployment Target rule

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2589]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

This rule will require a configuration to work.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

When bumping the deployment target of an app, it's common to leave unneeded code behind.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// would trigger
if #available(iOS 8.0, *) {            
}

@available(iOS 8.0, *) 
class A {}

// would not trigger
if #available(iOS 12.0, *) {            
}

@available(iOS 12.0, *) 
class A {}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Yes! `ios_deployment_target` and `macos_deployment_target`.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

This could go either way, because the rule shouldn't trigger if `ios_deployment_target` and `macos_deployment_target` are not set.
